graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/TopSecondDegreeByCountRequest.java

```diff
@@ -14,106 +14,54 @@
  * limitations under the License.
  */
 
-
 package com.twitter.graphjet.algorithms.counting;
 
-import java.util.Collections;
-import java.util.Map;
-import java.util.Set;
-
 import com.twitter.graphjet.algorithms.RecommendationRequest;
-import com.twitter.graphjet.algorithms.RecommendationType;
 import com.twitter.graphjet.algorithms.ResultFilterChain;
 import com.twitter.graphjet.hashing.SmallArrayBasedLongToDoubleMap;
-
 import it.unimi.dsi.fastutil.longs.Long2DoubleMap;
 import it.unimi.dsi.fastutil.longs.LongSet;
 
-public class TopSecondDegreeByCountRequest extends RecommendationRequest {
+/**
+ * Parent class of all requests using TopSecondDegreeByCount algorithm
+ */
+public abstract class TopSecondDegreeByCountRequest extends RecommendationRequest {
+
   private final Long2DoubleMap leftSeedNodesWithWeight;
-  private final Set<RecommendationType> recommendationTypes;
-  private final Map<RecommendationType, Integer> maxNumResultsByType;
   private final int maxSocialProofTypeSize;
-  private final int maxUserSocialProofSize;
-  private final int maxTweetSocialProofSize;
-  private final Map<RecommendationType, Integer> minUserSocialProofSizes;
   private final ResultFilterChain resultFilterChain;
-  private final Set<byte[]> socialProofTypeUnions;
 
   /**
-   * The constructor of TopSecondDegreeByCountRequest.
+   * Parent class constructor for requests to TopSecondDegreeByCountForTweet
```

Comment:
in this abstract class, we do not need to list every concrete class implementing it. otherwise every time we add a concrete class, we have to change the comment in the abstract class too.